operator/builtin/input/file/reader.go

```diff
@@ -172,7 +173,12 @@ func (r *Reader) emit(ctx context.Context, msgBuf []byte) error {
 		return fmt.Errorf("decode: %s", err)
 	}
 
-	e, err := r.fileInput.NewEntry(msg)
+	var e *entry.Entry
+	if r.fileInput.encoding.Encoding == encoding.Nop {
+		e, err = r.fileInput.NewEntry(msgBuf)
```

Comment:
Since we aren't using `msg`, we should avoid computing it.

Reply:
updated